File: app/src/main/java/com/sjn/stamp/constant/RecordType.java
package com.sjn.stamp.constant;

public enum RecordType {
    PLAY("play"),
    SKIP("skip"),
    START("start"),
    COMPLETE("complete");

    final public String mValue;

    public String getValue() {
        return mValue;
    }

    RecordType(String value) {
        mValue = value;
    }

    public static RecordType of(String value) {
        for (RecordType recordType : RecordType.values()) {
            if (recordType.getValue().equals(value)) return recordType;
        }
        return null;
    }

    public String getText() {
        return this.toString();
    }

}

File: app/src/main/java/com/sjn/stamp/constant/RecordType.kt
package com.sjn.stamp.constant

enum class RecordType(val value: String) {
    PLAY("play"),
    SKIP("skip"),
    START("start"),
    COMPLETE("complete");

    val text: String
        get() = this.toString()

    companion object {

        fun of(value: String): RecordType? {
            return RecordType.values().firstOrNull { it.value == value }
        }
    }

}